File: src/query_playbacks.ts
import { EMBY_URL, EMBY_DEVICE, urlWithParams } from './common.ts';

export async function queryPlaybacks(query: string) {
  const params = {
    aggregate_data: 'true',
    days: 365,
    end_date: '2024-10-30',
    filter_name: `${query}*`,
    ...EMBY_DEVICE,
  };

  const res = await fetch(
    urlWithParams(`${EMBY_URL}/emby/user_usage_stats/UserPlaylist`, params)
  );

  const data = await res.json();
  return data;
}

if (import.meta.main) {
  console.log(await queryPlaybacks('The Office (US)'));
}

File: src/query_playbacks.py
from src.common import EMBY_URL, EMBY_DEVICE, get_client


async def query_playbacks(query: str) -> dict:
    client = get_client()

    params = {
        "aggregate_data": "true",
        "days": 3650,
        "filter_name": f"{query}*",
        **EMBY_DEVICE,
    }

    res = await client.get(
        f"{EMBY_URL}/emby/user_usage_stats/UserPlaylist", params=params
    )
    return res.json()


if __name__ == "__main__":
    import asyncio

    print(asyncio.run(query_playbacks("Wonder Egg Priority")))
